pkg/kapp/resources/resources.go

```diff
@@ -305,9 +319,18 @@ func (c *ResourcesImpl) Patch(resource Resource, patchType types.PatchType, data
 	}
 
 	var patchedUn *unstructured.Unstructured
+	var dryRunOpt []string
+
+	if opts.DryRun {
+		dryRunOpt = []string{"All"}
+	}
 
 	err = util.Retry2(time.Second, 5*time.Second, c.isGeneralRetryableErr, func() error {
-		patchedUn, err = resClient.Patch(context.TODO(), resource.Name(), patchType, data, metav1.PatchOptions{})
+		patchedUn, err = resClient.Patch(context.TODO(), resource.Name(), patchType, data, metav1.PatchOptions{
+			FieldManager: c.opts.SSAFlags.FieldManagerName,
+			Force:        c.opts.SSAFlags.ForceParamValue(patchType),
```

Comment:
done.  I didn't test if Force flag on non Apply patch types breaks anything, so doing a safest option and passing it only on apply . It made logic a bit more convoluted, because we can't just pass `false` and every other case and has to pass `nil` instead.